### PR TITLE
Implements CleanupDerivatives change set persister handler and CleanupDerivativesJob

### LINKED ISSUE #147: Clean up files & derivatives on delete.

### PR DESCRIPTION
**Pull request by jrgriffiniii**
- Updating Valkyrie to https://github.com/samvera-labs/valkyrie/commit/dfcf1959b8399441a612d826b29cd1b44da10702
- Resolves #147 by extending the derivative services and implementing the CleanupDerivatives handler for the PlumChangeSetPersister

### CODE REVIEW

File: app/decorators/scanned_map_decorator.rb [hunk 1]
```diff
@@ -9,17 +9,20 @@ def members
...
   def geo_members
+    return [] if members.nil?
     members.select do |member|
       next unless member.respond_to?(:mime_type)
       ControlledVocabulary.for(:geo_image_format).include?(member.mime_type.first)
     end
   end
 
   def geo_metadata_members
+    return [] if members.nil?
```
Comment (tpendragon) [inline app/decorators/scanned_map_decorator.rb]: Can we just make `members` not nil?

File: app/derivative-services/image_derivative_service.rb [hunk 2]
```diff
@@ -77,7 +77,15 @@ def use
...
   end
 
-  def cleanup_derivatives; end
+  def cleanup_derivatives
+    deleted_files = []
+    image_derivatives = resource.file_metadata.select { |file| file.derivative? && file.mime_type.include?(image_mime_type) }
+    image_derivatives.each do |file|
+      storage_adapter.delete(id: file.id)
+      deleted_files << file.id
+      cleanup_derivative_metadata(derivatives: deleted_files)
```
Comment (tpendragon) [inline app/derivative-services/image_derivative_service.rb]: Should this be after the loop?

**Comment by hackmastera** [comment]
I'm not sure what the business logic is here. What derivatives do we want to clean up, and why? If there is a good place to add this info as a comment, that would probably be helpful to someone in the future.

**Comment by jrgriffiniii** [comment]
@hackmastera when you have the opportunity, please know that I have attempted to insert some documentation for the added methods and Classes (the build error seems to only be `Net::ReadTimeout` for CAS authentication, and is unrelated to this PR).

**Comment by hackmastera** [comment]
Aha, it's for cleaning up derivatives when a file has been deleted. That's very helpful, thanks @jrgriffiniii! I see now that it's in the issue but I think it's great to also have it in the code.

**Comment by jrgriffiniii** [comment]
You're welcome, and sorry for having overlooked documenting this.  For the future I will look to ensure that this not ignored.

File: app/decorators/ephemera_folder_decorator.rb [hunk 3]
```diff
@@ -140,4 +140,8 @@ def find_resource(resource_id)
     def controlled_value_for(value)
       value.present? && value.is_a?(Valkyrie::ID) ? find_resource(value) : value
     end
+
+    def find_members(resource:)
```
Comment (hackmastera) [inline app/decorators/ephemera_folder_decorator.rb]: maybe put `find_members` and `find_parents` in `Valkyrie::ResourceDecorator`?

**Comment by jrgriffiniii** [comment]
This is ready for another review please

**Comment by jrgriffiniii** [comment]
This has been rebased

**Pull request merged**